coalib/bearlib/aspects/collections.py

```diff
@@ -7,6 +8,21 @@ class aspectlist(list):
     """
 
     def __init__(self, seq=()):
+        """
+        Initialize a new aspectlist.
+
+        >>> from .root import Root
+        >>> from .Spelling import Spelling
+        >>> aspectlist([Root.Spelling.coalaCorrect, Root.Spelling.aspectsYEAH])
+        [<aspectclass '...coalaCorrect'>, <aspectclass '...aspectsYEAH'>]
+        >>> aspectlist(['coalaCorrect', 'aspectsYEAH'])
+        [<aspectclass '...coalaCorrect'>, <aspectclass '...aspectsYEAH'>]
+
+        :param seq: A sequence containing either aspectclass, aspectclass
+                    instance, or string of partial/full qualified aspect name.
+        """
+        if all(map((lambda x: isinstance(x, str)), seq)):
+            seq = map((lambda x: coalib.bearlib.aspects[x]), seq)
```

Comment:
Doesn't cover mixed cases like `aspectlist([Root.Spelling.coalaCorrect, 'aspectsYEAH'])`

Better `super().__init__(item if isaspect(item) else coalib.bearlib.aspects[item] for item in seq)`